cheatsheets/REST_Security_Cheat_Sheet.md

```diff
@@ -131,11 +131,27 @@ Services including script code (e.g. JavaScript) in their responses must be espe
 - Consider logging token validation errors in order to detect attacks.
 - Take care of log injection attacks by sanitising log data beforehand.
 
-## Security headers
+## Security Headers
 
-To make sure the content of a given resources is interpreted correctly by the browser, the server should always send the `Content-Type` header with the correct content type, and preferably the `Content-Type` header should include a charset. The server should also send the `X-Content-Type-Options: nosniff` [security header](https://owasp.org/www-project-secure-headers/) to make sure the browser does not try to detect a different `Content-Type` than what is actually sent (can lead to XSS).
+There are a number of [security related headers](https://owasp.org/www-project-secure-headers/) that can be returned in the HTTP responses to instruct browsers to act in specific ways. However, some of these headers are intended to be used with HTML responses, and as such may provide little or no security benefits on an API that does not return HTML.
 
-Additionally the server should send the `X-Frame-Options: deny` [security header](https://owasp.org/www-project-secure-headers/) to protect against drag'n drop clickjacking attacks in older browsers.
+The following headers should be included in all API responses:
+
+| Header | Rationale |
+|--------|-----------|
+| `Content-Security-Policy: frame-ancestors 'none'` | To protect against drag-and-drop style clickjacking attacks. |
```

Comment:
This is not required if HTML is not returned, or any "functionality" that could be triggered, so framing protection should be part of the second table, no?

Reply:
There's some interesting variants on clickjacking where you get to the user to select text and then drag it into a text box (that you control) so you can read the contents. That could still potentially be exploitable against an endpoint that returns JSON/text:

https://www.w3.org/Security/wiki/Clickjacking_Threats#Drag_and_drop_attacks